phys2bids/physio_obj.py

```diff
@@ -162,17 +251,103 @@ def ch_amount(self):
         """
         return len(self.timeseries)
 
-    def rename_channels(self, new_names, ch_trigger=None):
+    def __getitem__(self, idx):
+        """
+        Return a copy of the object with a sliced version of self.timeseries.
+
+        The slicing is based on the trigger. If necessary, computes a sort of
+        interpolation to get the right index in multifreq.
+        If the trigger was not specified, the slicing is based on the time instead.
+
+        Parameters
+        ----------
+        idx: int, tuple of int or slicer
+            indexes to use to slice the timeseries.
+
+        Returns
+        -------
+        BlueprintInput object
+            a copy of the object with the part of timeseries expressed by idx.
+
+        Raises
+        ------
+        IndexError
+            If the idx, represented as a slice, is out of bounds.
+
+        Notes
+        -----
+        If idx is an integer, it returns an instantaneous moment for all channels.
+        If it's a slicing, it always return the full slice. This means that
+        potentially, depending on the frequencies, BlueprintInput[1] and
+        BlueprintInput[1:2] might return different results.
+        """
+        sliced_timeseries = [None] * self.ch_amount
+        return_instant = False
+        trigger_length = len(self.timeseries[self.trigger_idx])
+
+        if not self.trigger_idx:
+            self.trigger_idx = 0
```

Comment:
```suggestion
        if not self.trigger_idx:
            self.trigger_idx = 0
        
        trigger_length = len(self.timeseries[self.trigger_idx])
```

Reply:
This should avoid getting errors in `self.timeseries[self.trigger_idx]`. See that if there's no `self.trigger_idx` that piece of code would raise an error.